scipy/integrate/_ivp/common.py

```diff
@@ -144,13 +144,21 @@ class OdeSolution:
     interpolants : list of DenseOutput with n_segments elements
         Local interpolants. An i-th interpolant is assumed to be defined
         between ``ts[i]`` and ``ts[i + 1]``.
+    alt_segment : boolean
```

Comment:
This part of PR is confusing to me.

Maybe we can get away with some internal testing and don't put this publicly?

I don't see it is used in the tests you've implemented. Also why change to "alt_segment" for the "BDF" and "LSODA" methods? If everything is correct it should not matter which segment to select.

Reply:
The idea here is that `BDF` and `LSODA` construct their interpolant differently from the `RK` methods. Specifically, at any given `t` during solver integration, `BDF` and `LSODA` use `y(t)` as a starting point, while other methods available in `scipy.integrate` use `y(t_old)`. Hence, in the current `main` branch, if one requests interpolation of the solution at `t` where `t` was an integration point, SciPy selects the interpolant segment for which `t` corresponded to `t` during the integration, as opposed to `t_old`. Hence, the value is by construction exact for `BDF` and `LSODA`, which is a problem for testing purposes as the accuracy of the interpolant at solver integration points is actually not tested. `alt_segment` fixes this by selecting the "non-trivial" segment for each method. It is not called directly by any of the tests in `test_ivp.py`. Rather, it is baked in `ivp.py` as part of the `OdeSolution` call in `solve_ivp`.